pkg/kubelet/cm/qos_container_manager_linux.go

```diff
@@ -82,8 +82,8 @@ func (m *qosContainerManagerImpl) GetQOSContainersInfo() QOSContainersInfo {
 func (m *qosContainerManagerImpl) Start(getNodeAllocatable func() v1.ResourceList, activePods ActivePodsFunc) error {
 	cm := m.cgroupManager
 	rootContainer := m.cgroupRoot
-	if !cm.Exists(rootContainer) {
-		return fmt.Errorf("root container %v doesn't exist", rootContainer)
+	if err := cm.Validate(rootContainer); err != nil {
+		return fmt.Errorf("root container %v invalid: %w", rootContainer, err)
```

Comment:
Isn't `rootContainer` a cgroup name? If so, the error message should probably mention it.